scylla-cql/src/types/serialize/serialize_value.rs

```diff
@@ -0,0 +1,29 @@
+use std::{any::Any, sync::Arc};
+
+use crate::{_macro_internal::Value, frame::response::result::ColumnType};
+
+pub trait SerializeCql {
+    fn preliminary_type_check(&self, typ: &ColumnType) -> bool;
```

Comment:
I think the preliminary type check should return `Result<(), SerializationError>` so that it can return some information about the error.